pkg/controller/node/status.go

```diff
@@ -44,12 +45,23 @@ func calculateStatus(pool *mcfgv1.MachineConfigPool, nodes []*corev1.Node) mcfgv
 	unavailableMachines := getUnavailableMachines(pool.Status.Configuration.Name, nodes)
 	unavailableMachineCount := int32(len(unavailableMachines))
 
+	degradedMachines := getDegradedMachines(pool.Status.Configuration.Name, nodes)
+	degradedReasons := make([]string, len(nodes))
+	for _, n := range degradedMachines {
+		reason, ok := n.Annotations[daemonconsts.MachineConfigDaemonReasonAnnotationKey]
+		if ok && reason != "" {
+			degradedReasons = append(degradedReasons, fmt.Sprintf("Node %s is reporting: %q", n.Name, reason))
```

Comment:
This is OK as is though in a larger cluster this could become a lot of text if e.g. 50 worker nodes are hitting the same issue.

Most issues I think though we can catch on the controller side before we try to roll out to each node.

Reply:
+1 on Colin's comment above. Couldn we just say X nodes report degraded and let user list them themselves?